zebra/kernel_netlink.c

```diff
@@ -316,7 +316,7 @@ static int netlink_socket(struct nlsock *nl, unsigned long groups,
 	int sock;
 	int namelen;
 
-	if (nl_family != NETLINK_ROUTE)
+	if (nl_family != NETLINK_ROUTE && nl_family != NETLINK_GENERIC)
```

Comment:
Do we really need this test?  This is a developer problem that they are passing in a wrong nl_family

Reply:
I removed this test.